src/Oscoin/Data/Tx.hs

```diff
@@ -120,3 +123,9 @@ validateTx Tx{..} =
     if verify txPubKey txMessage
     then Right ()
     else Left TxInvalidSignature
+
+
+evaluateBlock :: Evaluator c LegacyTxState (Tx c) DummyPayload
+evaluateBlock _beneficiary txs st = foldl' go ([], st) txs
+  where
+    go (output, st') tx = (output <> [txMessageContent tx], st')
```

Comment:
Dumb question -- any reason why this cannot be prepended rather than appended? I assume because that's linear ordering, i.e. the tx being evaluated depends from the context of what we have evaluated so far, and thus it needs to go at the end.

I do wonder if it would be more efficient to append at front and do a final `reverse` at the end -- that's usually the case for these kind of folds.

Reply:
The reason for this is indeed that we want the order of the outputs to match the order of the transactions. But you’re right that it would make sense to prepend and reverse.